vlm/src/main/scala/geotrellis/contrib/vlm/GeotrellisRasterSource.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2018 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.contrib.vlm
+
+import geotrellis.contrib.vlm._
+import geotrellis.contrib.vlm.geotiff._
+import geotrellis.vector._
+import geotrellis.proj4._
+import geotrellis.raster._
+import geotrellis.raster.reproject.Reproject
+import geotrellis.raster.resample.ResampleMethod
+import geotrellis.raster.io.geotiff.{GeoTiffMultibandTile, MultibandGeoTiff, OverviewStrategy}
+import geotrellis.raster.io.geotiff.reader.GeoTiffReader
+import geotrellis.spark.{LayerId, Metadata, SpatialKey, TileLayerMetadata}
+import geotrellis.spark.io._
+import geotrellis.spark.io.avro.codecs._
+import geotrellis.raster.{MultibandTile, Tile}
+
+/**
+  * Note: GeoTrellis AttributeStore does not store the band count for the layers by default,
+  *       thus they need to be provided from application configuration.
+  *
+  * @param uri geotrellis catalog uri
+  * @param layerId source layer from above catalog
+  * @param bandCount number of bands for each tile in above layer
+  */
+case class GeotrellisRasterSource(uri: String, layerId: LayerId, bandCount: Int = 1) extends RasterSource {
```

Comment:
For consistency this should move into its own package.
I propose `goetrellis.contrib.vlm.avro`

Reply:
Sure. Should I also move `TestCatalog.scala` into `goetrellis.contrib.vlm.avro` then?

Reply:
Thats a good point, it should go there.